sdk-workflows/pom.xml

```diff
@@ -51,7 +51,7 @@
     <dependency>
       <groupId>com.microsoft</groupId>
       <artifactId>durabletask-client</artifactId>
-      <version>1.1.1</version>
+      <version>1.5.0</version>
```

Comment:
does upgrading to this version enable us to skip over the breaking change mentioned here https://github.com/dapr/java-sdk/issues/944?

Reply:
The break change will be introduced with this upgrade. 
I would be able to help work on the workflow example. 

The reason we introduce this change to `continueAsNew` is that we hear many customers say it's confusing as they expect it will immediately restart the orchestration instead of continuing to run the code after it. Just like the `continue` in a for loop, no one expects code after `continue` to be run in current loop. 
Hopefully, this won't raise any concerns, but let me know if we want to approach this differently.

Reply:
I can see why that would cause confusion... Will we need to update our workflow examples and code as a result of this change in behavior, as a result of the upgrade? If yes, I vote to pull the upgrade out of this PR to a new one with the relevant changes, if not, then it's probably fine as is.

Reply:
The deterministic UUID feature is added to `durabletask-client` in this version v1.5.0, so it's needed for this PR. I can try to create another PR to update the workflow example, at least make it work with this breaking change. Or do you think including the example update in this PR is better? 

Besides the example update, I don't think there is any other code change required, since we are calling `continueAsNew ` API of the `durabletask-client` at the end. But please let me know if I miss something.

Reply:
oh I see, I think the example can be updated in a follow up PR. @artursouza do you agree?

Reply:
Waiting on this to merge.

Reply:
I'm a little bit confused about the CI error, I don't think this change is related to those errors. Also for some jdk version it does pass. Could they be intermittent errors? Is there something I missed? Thanks.

Reply:
Merging this after tests pass.

Reply:
I think tests passed.